cmd/shutdown.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"time"
+)
+
+const (
+	shutdownTimeout = 10 * time.Second
+)
+
+//GracefulShutdown implements releasing all resouces it got from system, finish all request handling and return responses when service stopping.
+func gracefulShutdown(done chan<- bool, components []io.Closer) error {
+	ctx, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
+
+	var errTimeout error
+
+	go func() {
+		for {
+			errTimeout = ctx.Err()
```

Comment:
You can use ctx.Done() instead of the `for` loop. Two problems: race on `errTimeout` and high cpu usage

Reply:
fixed